fineract-provider/src/integrationTest/java/org/apache/fineract/integrationtests/SchedulerJobsTestResults.java

```diff
@@ -724,6 +724,59 @@ public void testApplyPenaltyForOverdueLoansJobOutcome() throws InterruptedExcept
 
     }
 
+    @Test
+    public void testAvoidUnncessaryPenaltyWhenAmountZeroForOverdueLoansJobOutcome() throws InterruptedException {
```

Comment:
I'm unable to say if this is functionally correct - would someone else be able to review and confirm it is?

Reply:
@awasum @Anh3h do you understand more about the functionality here than I do and can you confirm this new test is functionally correct?

Reply:
I can throw some light into this. 
It is creating a loan with a very small interest rate 0.000001%.
It is verifying that the `totalChargesDue` is 0 for the loan ( which is also the feature currently as 0+ 0 + 0 is 0).
I am then calling the api `/loans/{id}?associations=charges` and then verifying that there are no charges created for this loan. This actually tests correctly what I have done.